hedera-node/src/main/java/com/hedera/services/fees/calculation/consensus/txns/UpdateTopicResourceUsage.java

```diff
@@ -49,15 +49,19 @@ public boolean applicableTo(TransactionBody txn) {
     public FeeData usageGiven(TransactionBody txn, SigValueObj sigUsage, StateView view) throws InvalidTxBodyException {
         try {
             MerkleTopic merkleTopic = view.topics().get(MerkleEntityId.fromTopicId(txn.getConsensusUpdateTopic().getTopicID()));
-            long rbsIncrease = getUpdateTopicRbsIncrease(
-                    txn.getTransactionID().getTransactionValidStart(),
-                    JKey.mapJKey(merkleTopic.getAdminKey()),
-                    JKey.mapJKey(merkleTopic.getSubmitKey()),
-                    merkleTopic.getMemo(),
-                    merkleTopic.hasAutoRenewAccountId(),
-                    lookupExpiry(merkleTopic),
-                    txn.getConsensusUpdateTopic());
+            if(merkleTopic != null && merkleTopic.hasAdminKey()) {
```

Comment:
1. #1504 only mentions missing topic, not immutable one. This is a mismatch which should be fixed here or in the issue. I think fixing in the issue makes more sense because the ram bytes won't change due to the topic immunity.
2. After this check, I believe we will have only `DecoderException` from the `JKey.mapJKey`, so the `catch` part should be changed accordingly.
3. After this check, the `lookupExpiry(merkleTopic)` on (new) line 59 is not applicable any more. We should be able to directly return the expiration timestamp from the topic. The try catch in the original helper function was overdone too. It should have been a simple if then else.

@tinker-michaelj, please help confirm my arguments. Thanks!

Reply:
Agree with 1 and 2. But 3, I think the NPE check is on `merkleTopic.getExpirationTimestamp()` rather than`merkleTopic`

Reply:
No, here is its code snippet
```
	public RichInstant getExpirationTimestamp() {
		return hasExpirationTimestamp() ? expirationTimestamp : new RichInstant();
	}
```

Reply:
yup.

Reply:
Updated the issue too along with the changes requested. Thanks Quan.

Reply:
Realized, that we should also catch NPE in point 2.

Reply:
Which function may cause NPE?

Reply:
TransactionBody being null / stateView being null. We have a unit test for this.